scripts/wsrep_sst_mariabackup.sh

```diff
@@ -522,9 +522,13 @@ wait_for_listen()
     local PORT=$1
     local ADDR=$2
     local MODULE=$3
+    if ! which lsof > /dev/null; then
+       wsrep_log_error "lsof tool not found in PATH! Make sure you have it installed."
```

Comment:
We can't introduce a new dependencies to GA-releases, so better solution would be check is ss command found if not then try lsof and if both are missing fail.

Reply:
I used lsof here because I saw 'lsof' is widely used in other wsrep scripts. On FreeBSD, the best tool to check if a port is listening is to use sockstat.  I think it would be better to check if ```[ "$OS" == "FreeBSD" ]``` as in wsrep_sst_rsync.

Reply:
Why is `sockstat` the best? Does it implement something `lsof -sTCP:LISTEN -i TCP:${PORT} -a -c nc -c socat -F c` doesn't? if you `truss` the implementation does it result in something meaningfully different? Relying on lsof people to implement a consistent interface across operating systems is their speciality and reduces the maintenance test burden on the mariadb side.

Reply:
Because ```sockstat``` comes with the system, while lsof is a thirdparty tool. I think the less dependency, the better.

Reply:
@janlindstrom @grooverdan `lsof` would be overkill on BSD systems, they all have `sockstat` (since forever) which provides what is needed. On any BSD system `lsof` would be an additional dependency. I must say I like the selector from `ss` 😄, pitty it doesn't select process names.

I've replaced `lsof` with `sockstat` in an earlier MDEV-10432 for MariaDB. The wsrep_sst_common.sh script was converted to POSIX compatibility for that too.

```
@@ -532,7 +532,11 @@ wait_for_listen()
     local MODULE=$3
     for i in {1..50}
     do
-        ss -p state listening "( sport = :$PORT )" | grep -qE 'socat|nc' && break
+        if [ "$OS" = "FreeBSD" ];then
+            sockstat -46lp $PORT | grep -qE "^[^ ]* *(socat|nc) *[^ ]* *[^ ]* *[^ ]* *[^ ]*:$PORT" && break
+        else
+            ss -p state listening "( sport = :$PORT )" | grep -qE 'socat|nc' && break
+        fi
         sleep 0.2
     done
     echo "ready ${ADDR}/${MODULE}//$sst_ver"
```
The grep is elaborate to make sure only the local is selected, otherwise it would also select the remote port.